### ISSUE TITLE
ERROR: TypeError: Cannot read properties of null (reading 'replace')

### ISSUE THREAD

**Issue by wiibe** [sitespeedio/sitespeed.io#3617]
Hi guys i am walking in to a problem when i am outputting budget to a junit xml file.

### What are you trying to accomplish

I'm trying to output the performance budget to an junit file and i am getting the following error:

```shell
[2022-04-01 12:52:52] INFO: Budget: 2 working, 0 failing tests and 0 errors
[2022-04-01 12:52:52] ERROR: TypeError: Cannot read properties of null (reading 'replace')
    at Object.exports.writeJunit (/usr/src/app/lib/plugins/budget/junit.js:16:26)
    at Object.processMessage (/usr/src/app/lib/plugins/budget/index.js:110:21)
    at /usr/src/app/lib/core/queueHandler.js:213:32
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain [as _onImmediate] (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at processImmediate (node:internal/timers:466:21)
[2022-04-01 12:52:53] INFO: HTML stored in /sitespeed.io/report
```

### Complete output

```shell
Chromium 97.0.4692.20 Ubuntu 20.04
Mozilla Firefox 98.0.2
[2022-04-01 13:41:36] INFO: Versions OS: linux 5.10.104-linuxkit nodejs: v16.14.2 sitespeed.io: 23.7.0 browsertime: 15.4.0 coach: 7.1.0
[2022-04-01 13:41:36] INFO: Running tests using Chrome - 1 iteration(s)
[2022-04-01 13:41:36] INFO: Testing url https://www.sitespeed.io iteration 1
[2022-04-01 13:41:46] INFO: Take after page complete check screenshot
[2022-04-01 13:41:46] INFO: Take cumulative layout shift screenshot
[2022-04-01 13:41:47] INFO: Take largest contentful paint screenshot
[2022-04-01 13:41:50] INFO: Get visual metrics from the video
[2022-04-01 13:41:51] INFO: https://www.sitespeed.io 11 requests, TTFB: 134ms, firstPaint: 219ms, firstVisualChange: 235ms, FCP: 219ms, DOMContentLoaded: 223ms, LCP: 302ms, CLS: 0.0004, TBT: 0ms, Load: 274ms, speedIndex: 253ms, visualComplete85: 294ms, lastVisualChange: 294ms
[2022-04-01 13:41:52] INFO: Running tests using Chrome - 1 iteration(s)
[2022-04-01 13:41:52] INFO: Testing url https://www.sitespeed.io/404 iteration 1
[2022-04-01 13:42:01] INFO: Take after page complete check screenshot
[2022-04-01 13:42:02] INFO: Take cumulative layout shift screenshot
[2022-04-01 13:42:02] INFO: Take largest contentful paint screenshot
[2022-04-01 13:42:05] INFO: Get visual metrics from the video
[2022-04-01 13:42:06] INFO: https://www.sitespeed.io/404 3 requests, TTFB: 317ms, firstPaint: 337ms, firstVisualChange: 752ms, FCP: 684ms, DOMContentLoaded: 331ms, LCP: 734ms, CLS: 0, TBT: 0ms, Load: 700ms, speedIndex: 752ms, visualComplete85: 752ms, lastVisualChange: 752ms
[2022-04-01 13:42:06] INFO: Failing budget Speed Index for 404 with value 752 ms max limit 500 ms
[2022-04-01 13:42:06] INFO: Budget: 1 working, 1 failing tests and 0 errors
[2022-04-01 13:42:06] ERROR: TypeError: Cannot read properties of null (reading 'replace')
    at Object.exports.writeJunit (/usr/src/app/lib/plugins/budget/junit.js:16:26)
    at Object.processMessage (/usr/src/app/lib/plugins/budget/index.js:110:21)
    at /usr/src/app/lib/core/queueHandler.js:213:32
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain [as _onImmediate] (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at processImmediate (node:internal/timers:466:21)
[2022-04-01 13:42:08] INFO: HTML stored in /sitespeed.io/report
```

### Specs

- Docker Desktop 4.6.1 (76265) 
- macOs Monterey 12.3.1
- MacBook Pro 16-inch 2021
- Chip Apple M1 Pro

- Also tested on a iIntel chip Mac with the same problems

### Relevant information

I found out that it is working in version **23.1.0** and when i am using version **23.2.0** or above it is broken.
I also found out it has something to do with the "Aliases" because when i remove them it is working again.
It only occurs when we try to output to junit, json and tap are working fine.

### How to reproduce

See attached below the simplified config i am using:

```shell
docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io --config config.json --outputFolder report urls.txt
```

config.json
```json
{
  "multi": false,
  "budget": {
    "configPath": "budget.json",
    "output": "junit"
  },
  "screenshot": {
    "type": "jpg"
  },
  "metrics": {
    "filter": []
  },
  "browsertime": {
    "iterations": 1,
    "viewPort": "1280x720",
    "browser": "chrome",
    "visualMetrics": true,
    "video": false
  }
}
```

budget.json
```json
{
  "budget": {
    "Homepage": {
      "timings": {
        "SpeedIndex": 1500
      }
    },
    "404": {
      "timings": {
        "SpeedIndex": 500
      }
    }
  }
}
```

urls.txt
```txt
https://www.sitespeed.io Homepage
https://www.sitespeed.io/404 404
```

Hope you guys can help me out to sort this issue.

**Comment by soulgalore** [comment]
Hi @wiibe thanks for creating the issue, I'll have a look ASAP.

**Comment by soulgalore** [comment]
Thank you @wiibe for taking the time and make a really good bug report! I caused the bug when I implemented some better support for alias in the budget file. It's fixed in #3620 and I hope to release the later today.

**Comment by soulgalore** [comment]
The release will be tomorrow then I can push a couple of other changes.

**Comment by soulgalore** [comment]
Released in 24.0m @wiibe please let me know when you have tested it!

**Comment by wiibe** [comment]
@soulgalore Thanks for the quick response and quick fix.

I have tested it in 2 of my projects and it is working as expected now!

Thanks again, and thanks for this great project !
